modules/tensor_mechanics/include/postprocessors/CriticalTimeStep.h

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ElementPostprocessor.h"
+#include "GuaranteeConsumer.h"
+#include "ComputeStressBase.h"
+
+// Forward Declarations
+class CriticalTimeStep;
+
+template <>
+InputParameters validParams<CriticalTimeStep>();
+
+class CriticalTimeStep : public ElementPostprocessor, public GuaranteeConsumer
+{
+public:
+  CriticalTimeStep(const InputParameters & parameters);
+
+  virtual void initialize() override;
+  virtual void execute() override;
+  virtual void initialSetup() override;
+
+  virtual Real getValue() override;
+  virtual void threadJoin(const UserObject & y) override;
+
+protected:
+  const MaterialProperty<Real> & _mat_dens;
+
+  const MaterialProperty<Real> & _effective_stiffness;
+
+  const Real & _factor;
+
+  Real _total_size;
+  int _elems;
```

Comment:
Thanks you for your comment. I will use _unsigned long int_ for this, which has 4.2 billion capacity.

Also, moose/framework/src/postprocessors/AverageElementSize.h uses 'int _elems'. Should I make changes there too?

Reply:
The right type is "dof_id_type"

Reply:
Yes, anything dealing with numbers of elements should be `dof_id_type`, or if it's related to indicies in a container `std::size_t`. C++11 gives us a lot of tools "auto" and "decltype" that you can use in many places to avoid picking the wrong type explicitly.